lint-staged.config.mjs

```diff
@@ -0,0 +1,23 @@
+/**
+ * @param {string[]} filenames
+ * @returns {string}
+ */
+const filenamesToAjvOption = (filenames) => filenames.map((filename) => `-d ${filename}`).join(' ');
+
+/** @satisfies {import('lint-staged').Config} */
```

Comment:
We should run eslint and typecheck here as well IMO.

Reply:
After this PR is merged, I will submit a separate PR related to ESLint. How about we handle it together at that time?